cfme/tests/middleware/test_middleware_provider.py

```diff
@@ -10,11 +16,89 @@
     pytest.mark.uncollectif(lambda: current_version() < '5.7'),
 ]
 
+pytest_generate_tests = testgen.generate(testgen.middleware_providers, scope="function")
+
+
+@pytest.mark.usefixtures('has_no_middleware_providers')
+def test_provider_add_with_bad_credentials(provider):
+    """ Tests provider add with bad credentials"""
+    provider.credentials['default'] = provider.Credential(
+        principal='bad',
+        secret='reallybad'
+    )
+    with error.expected('Credential validation was not successful: Invalid credentials'):
+        provider.create(validate_credentials=True)
+
+
+def test_add_cancelled_validation():
+    """Tests that the flash message is correct when add is cancelled."""
+    prov = HawkularProvider()
+    prov.create(cancel=True)
+    flash.assert_message_match('Add of new Middleware Provider was cancelled by the user')
+
+
+def test_password_mismatch_validation():
+    """ Tests password mismatch check """
+    prov = HawkularProvider()
+    cred = prov.Credential(
+        principal='bad',
+        secret=fauxfactory.gen_alphanumeric(5),
+        verify_secret=fauxfactory.gen_alphanumeric(6)
+    )
+    sel.force_navigate('{}_provider_new'.format(prov.page_name))
+    fill(prov.properties_form, {"type_select": "Hawkular"})
+    fill(prov.properties_form, {"hostname_text": "test"})
+    fill(prov.properties_form, {"name_text": "test"})
+    fill(cred.form, cred)
```

Comment:
This is not how fill is intended to be used. You shall pass all the values to fill, it goes through the form and fills the values then.